src/java/org/apache/cassandra/io/util/DataInputPlus.java

```diff
@@ -30,11 +30,31 @@
 @Shared(scope = SIMULATION)
 public interface DataInputPlus extends DataInput
 {
+    /**
+     * Read a 64-bit integer back. This assumes it was originally written using writeVInt
+     * and uses zigzag encoding.
+     * @return
+     * @throws IOException
+     */
     default long readVInt() throws IOException
     {
         return VIntCoding.readVInt(this);
     }
 
+    /**
+     * Read up to a 32-bit integer back, throwing IllegalArgumentException if
+     * the vint read is out of range for a 32-bit integer.
+     *
+     * This method assumes the integer was originally written using writeVInt32
+     * and uses zigzag encoding.
```

Comment:
So in general, I don't care a ton how we document return stuff here, as long as it's consistent across this class. ex. We could replace the above w/ just...
```
@return a 32-bit integer, assumed to originally have been written using {@link DataOutputPlus#writeVInt32(int)}
```
...assuming you like that format.

Reply:
I think it's possibly too verbose? The return is just an integer. How it was encoded is more a matter of how the method works and where it gets it from? Seems like we shouldn't provide that information twice.

Reply:
Yeah, I was more suggesting replacing the description, not providing the information twice. Anyway, just removing the `@return` is fine w/ me.